contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -113,8 +133,11 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
             );
         }
 
+        uint256 scaledAmount = _scaleTokens(value, false);
```

Comment:
If `multiplyOnReceive` is true, should we require that the value here is divisible by `tokenMultiplier`? In practice it should be only very minor amounts that get lost due to integer division, but in theory those could be worth a significant amount. 

We're not able to handle the non-divisible amount case when receiving either way, but it may make sense to handle it here when sending.

Reply:
I think that would be over-restrictive. The max we can lose in a transfer is less than one of the smallest divisible unit on one of the chains. If there's ever a situation where that is worth a non-negligible amount, checks could be added user-side.

Reply:
Restricting transfer amounts to only valid values is a good restriction to have IMO. Otherwise we're making an additional assumption (the difference is worth a significant amount), which is likely to be true in the vast majority of use cases but feasibly not all. 

As is, it's also possible for `scaledAmount` to be 0, which I think would be an invalid case.

Reply:
I also think we should only transfer amounts that are divisible, but think we can handle that logic in the contract to not restrict the passed in transfer amount argument. If the user passes an amount that is not divisible, we `removeDust`, meaning we return any amount that is non divisible back to the sender, and bridge the rest. LayerZero should have a similar concept in `OFT`s.

Reply:
I'm not sure if returning anything to the sender is a good idea here. Since we're dealing with native tokens, any extra operations are going to cost us more gas, and those operations are going to use up more gas than we're returning unless the multipliers are VERY large (Greater than 10^12).